internal/common/objects/BikerAgent.go

```diff
@@ -0,0 +1,59 @@
+package baseextendedagent
+
+/*
+	BikerAgent is the implentation of IBikerAgent. All extended biker agents must be composed of a BikerAgent object
+*/
+
+import (
+	baseagent "github.com/MattSScott/basePlatformSOMAS/BaseAgent"
+	"github.com/google/uuid"
+	// "SOMAS2023/internal/common/utils"
+)
+
+type BikerAgent struct {
+	colour Colour
+	forces Forces
+	position [2]float64
+}
+
+// generates a new BikerAgent instance with a randomised ID
+NewBikerAgent [T IBikerAgent [ T ]]() ∗ BikerAgent [ T ] {
+	return &BaseAgent [ T] {
+		id uuid.ID
+	}
+}
```

Comment:
Biker agent doesn't use generics. Since this is the 'intermediate' class, you need to compose the `baseagent.BaseAgent[IBikerAgent]`

This should then return `*BikerAgent` to be composed by users